config/cmake/modules/FindSLEPc.cmake

```diff
@@ -0,0 +1,278 @@
+# - Try to find SLEPC
+# Once done this will define
+#
+#  SLEPC_FOUND        - system has SLEPc
+#  SLEPC_INCLUDE_DIR  - include directories for SLEPc
+#  SLEPC_LIBARIES     - libraries for SLEPc
+#  SLEPC_DIR          - directory where SLEPc is built
+#  SLEPC_VERSION      - version of SLEPc
+#  SLEPC_VERSION_MAJOR - First number in SLEPC_VERSION
+#  SLEPC_VERSION_MINOR - Second number in SLEPC_VERSION
+#  SLEPC_VERSION_SUBMINOR - Third number in SLEPC_VERSION
+#
+# Assumes that PETSC_DIR and PETSC_ARCH has been set by
+# alredy calling find_package(PETSc)
+
+#=============================================================================
+# Copyright (C) 2010-2012 Garth N. Wells, Anders Logg and Johannes Ring
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in
+#    the documentation and/or other materials provided with the
+#    distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#=============================================================================
```

Comment:
This has to be kept in mind. Might be necessary or not wanted for certain customer situations. Ping @acfisher @tzanio.

Reply:
I am not very familiar with cmake so I reused an existing file from another project. The FindPETSc.cmake has a similar header (this might be a separate issue). If you prefer I could write something based on https://gitlab.com/slepc/slepc/-/issues/19, but it would depend on pkg-config.

Reply:
What should I do with this?

Reply:
This whole file can probably be simplified a lot by using something like this: https://github.com/mfem/mfem/blob/cc3a4475d298df742de0a98d9c4586c3fd02ef82/config/cmake/modules/FindSTRUMPACK.cmake#L12-L36